### PR TITLE
Add missing documentation

### PR DESCRIPTION
**Pull request by AndreasBurger**

### CODE REVIEW

File: product/scanning.py [hunk 1]
```diff
@@ -22,16 +22,50 @@
...
+                doc='''
+                    trigger a (re)scan of container images. Images will be uploaded if they are not
+                    already present.
+                ''',
+                type=str,
+            ),
+            AttributeSpec.optional(
+                name=cls.FORCE_UPLOAD.value,
+                default=False,
+                doc='upload the container images. This will :strong:`always` upload the images.',
```
Comment (ccwienk) [inline product/scanning.py]: nit: how about just: "always upload"

File: product/scanning.py [hunk 2]
```diff
@@ -22,16 +22,50 @@
...
+                    upload the container images. This will :strong:`not` upload the images if they
+                    are already present. Scans will be triggered only for images that are uploaded.
+                ''',
+                type=str,
+            ),
+            AttributeSpec.optional(
+                name=cls.RESCAN.value,
+                default=False,
+                doc='''
+                    trigger a (re)scan of container images. Images will be uploaded if they are not
```
Comment (ccwienk) [inline product/scanning.py]: nit: unless they are..

Comment (ccwienk) [reply product/scanning.py]: also: what precisely was the difference between UPLOAD_IF_CHANGED and RESCAN again? I don't remember. and I do not understand the difference from the (new) documentation tbh

File: product/scanning.py [hunk 3]
```diff
@@ -22,16 +22,50 @@
...
     UPLOAD_IF_CHANGED = 'upload_if_changed'
     RESCAN = 'rescan'
     FORCE_UPLOAD = 'force_upload'
 
+    @classmethod
+    def _attribute_specs(cls):
+        return (
+            AttributeSpec.optional(
+                name=cls.UPLOAD_IF_CHANGED.value,
+                default=True,
```
Comment (ccwienk) [inline product/scanning.py]: i'm still not happy with the mis-use of the 'default' attribute for enums

File: concourse/model/traits/release.py [hunk 4]
```diff
@@ -67,10 +85,11 @@ class ReleaseNotesPolicy(enum.Enum):
     ),
     AttributeSpec.optional(
         name='release_notes_policy',
-        default=ReleaseNotesPolicy.DEFAULT,
+        default=ReleaseNotesPolicy.DEFAULT.value,
```
Comment (ccwienk) [inline concourse/model/traits/release.py]: nit: would it not be better to use the "real" value as default value (the enum instance) and do the conversion in the getter exposing the attr? this holds also true for other occurrences..

File: concourse/model/traits/image_scan.py [hunk 5]
```diff
@@ -81,6 +81,7 @@ def exclude_image_references(self):
         name='processing_mode',
         default='upload_if_changed',
         doc='Protecode processing mode', # todo: document allowed values
```
Comment (ccwienk) [inline concourse/model/traits/image_scan.py]: nit: todo is no longer required ;-)

File: product/scanning.py (hunk 2 above)
Comment (AndreasBurger) [reply product/scanning.py]: UPLOAD_IF_CHANGED uploads and, thus, rescans only those images whose image descriptor changed. RESCAN causes all given container images to be rescanned, uploading those that are not available anymore.

I've tried to clarify it a bit.

File: product/scanning.py (hunk 3 above)
Comment (AndreasBurger) [reply product/scanning.py]: Repalced with a lesser mis-use. Probably worth discussing further tomorrow morning.

Comment (AndreasBurger) [reply product/scanning.py]: After another look: I guess it's actually used as intended now.

**Pull request merged**